src/utils/twoFactor.js

```diff
@@ -0,0 +1,181 @@
+    const request = {
+        receiver_id: account.accountId,
+        actions: [addKeyAction(account, publicKey, contractId, fullAccess)]
+    }
+    return twoFactorRequest(wallet, account, request)
+}
+
+export const twoFactorRemoveKey = (wallet, account, publicKey) => {
+    const request = {
+        receiver_id: account.accountId,
+        actions: [deleteKeyAction(publicKey)]
+    }
+    return twoFactorRequest(wallet, account, request)
+}
+/********************************
+Deploys 2/3 multisig contract with keys from contract-helper and localStorage
+@todo check account has enough near to actually do this
+********************************/
+export const twoFactorDeploy = async (wallet) => {
+    const accountData = await wallet.loadAccount()
+    // get multisig contract to deploy
+    const contractBytes = new Uint8Array(await fetch('/multisig.wasm').then((res) => res.arrayBuffer()))
+    // get account
+    const { accountId } = accountData
+    const account = wallet.getAccount(accountId)
+    // 1. get localstorage keys
+    const accountKeys = await account.getAccessKeys();
+    // 2. get recovery method keys
+    const recoveryMethods = await wallet.getRecoveryMethods()
+    const recoveryKeysED = recoveryMethods.data.map((rm) => rm.publicKey)
+    // 3. get recovery keys that are NOT seed phrases && NOT null publicKey
+    const fak2lak = recoveryMethods.data.filter(({ kind, publicKey }) => kind !== 'phrase' && publicKey !== null).map((rm) => toPK(rm.publicKey))
+    // 4. push localStorage keys that are NOT recovery keys
+    fak2lak.push(...accountKeys.filter((ak) => !recoveryKeysED.includes(ak.public_key)).map((ak) => toPK(ak.public_key)))
+    // these are the keys we need to convert to limited access keys
+    // 5. get the server public key for wallet accountId (confirmOnlyKey)
+    const getAccessKey = await wallet.postSignedJson('/2fa/getAccessKey', {
+        accountId
+    })
+    if (!getAccessKey || !getAccessKey.success || !getAccessKey.publicKey) {
+        console.log('error getting publicKey from contract-helper')
+        return
+    }
+    const confirmOnlyKey = toPK(getAccessKey.publicKey)
+    const newArgs = new Uint8Array(new TextEncoder().encode(JSON.stringify({ 'num_confirmations': 2 })));
+    const actions = [
+        // delete FAKs and add them as LAKs
+        ...fak2lak.map((pk) => nearApiJs.transactions.deleteKey(pk)),
+        ...fak2lak.map((pk) => nearApiJs.transactions.addKey(pk, nearApiJs.transactions.functionCallAccessKey(accountId, METHOD_NAMES_LAK, null))),
+        // confirmOnlyKey
+        nearApiJs.transactions.addKey(confirmOnlyKey, nearApiJs.transactions.functionCallAccessKey(accountId, METHOD_NAMES_CONFIRM, null)),
+        // deploy and initialize contract
+        nearApiJs.transactions.deployContract(contractBytes),
+        nearApiJs.transactions.functionCall('new', newArgs, 10000000000000, '0'),
+    ]
+    console.log('deploying multisig contract')
```

Comment:
Is this debugging log or intended to stay? If intended to stay – makes sense to log account id.

Reply:
I don't think it harms. Added accountId.